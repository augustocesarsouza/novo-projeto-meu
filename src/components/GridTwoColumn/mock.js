// eslint-disable-next-line import/no-anonymous-default-export
export default {
  title: "Grid two columns",
  text: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Assumenda soluta illum accusantium unde ullam, cum recusandae temporibus enim iure reprehenderit, repellendus sit vitae quidem dolorum ad voluptate eaque cupiditate. Atque!`,
  srcImg: "assets/imagens/javascript.svg",
};
