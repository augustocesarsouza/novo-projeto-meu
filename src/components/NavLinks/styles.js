import styled, { css } from "styled-components";
export const Container = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row wrap;

    @media ${theme.media.lteMedium} {
      flex-wrap: column wrap;
      align-content: center;
    }
  `}
`;
