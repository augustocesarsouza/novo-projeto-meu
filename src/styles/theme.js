export const theme = {
  colors: {
    mainBg: "red",
    secondaryBg: "blue",
    green: "green",
  },
  fonts: {},
  spacing: {},
};
