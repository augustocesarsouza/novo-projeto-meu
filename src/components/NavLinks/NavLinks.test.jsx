import { screen } from "@storybook/testing-library";
import { NavLinks } from ".";
import { renderTheme } from "../../styles/render-theme";
import { theme } from "../../styles/theme";

import mock from "./mock";

describe("<NavLinks />", () => {
  it("should render links", () => {
    renderTheme(<NavLinks links={mock} />);
    expect(screen.getAllByRole("link")).toHaveLength(mock.length);
  });

  it("should not render links", () => {
    renderTheme(<NavLinks />);
    expect(screen.queryAllByText(/links/i)).toHaveLength(0);
  });

  it("should render links", () => {
    renderTheme(<NavLinks links={mock} />);
    expect(screen.getByText(/link 10/i).parentElement).toHaveStyleRule("flex-wrap", "column wrap", {
      media: theme.media.lteMedium,
    });
  });

  it("should match snapshot", () => {
    renderTheme(<NavLinks links={mock} />);
    expect(screen.getAllByRole("link")).toMatchSnapshot();
  });
});
