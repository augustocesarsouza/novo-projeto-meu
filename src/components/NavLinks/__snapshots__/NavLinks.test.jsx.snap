// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLinks /> should match snapshot 1`] = `
.c1 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0A1128;
  position: relative;
}

.c1::after {
  content: "";
  position: absolute;
  bottom: 0.76rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #DC143C;
  -webkit-transition: :all 300ms ease-in-out;
  transition: :all 300ms ease-in-out;
}

.c1:hover::after {
  left: 25%;
  width: 50%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
  }
}

<nav
  aria-label="Main menu"
  class="c0"
>
  <a
    class="c1"
    href="#target 1"
    target="_self"
  >
    Link 1
  </a>
  <a
    class="c1"
    href="#target 2"
    target="_self"
  >
    Link 2
  </a>
  <a
    class="c1"
    href="#target 3"
    target="_self"
  >
    Link 3
  </a>
  <a
    class="c1"
    href="#target 4"
    target="_self"
  >
    Link 4
  </a>
  <a
    class="c1"
    href="#target 5"
    target="_self"
  >
    Link 5
  </a>
  <a
    class="c1"
    href="#target 6"
    target="_self"
  >
    Link 6
  </a>
  <a
    class="c1"
    href="#target 7"
    target="_self"
  >
    Link 7
  </a>
  <a
    class="c1"
    href="#target 8"
    target="_self"
  >
    Link 8
  </a>
  <a
    class="c1"
    href="#target 9"
    target="_self"
  >
    Link 9
  </a>
  <a
    class="c1"
    href="#target 10"
    target="_self"
  >
    Link 10
  </a>
</nav>
`;
